Tree.cs

```diff
@@ -0,0 +1,91 @@
+﻿using System;
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Diagnostics;
+
+namespace csMTG
+{
+   public class Tree
+    {
+        // Define the attributes
+        int id { get; set; } = 0;
+        //The key is the parent's id, and the values are a list of children
+        protected Dictionary<int, List<int>> parent = new Dictionary<int, List<int>>();

```

Comment:
parent need to be replaced by children (parentId -> [chidrenId])

Create a parent attribute:
 protected Dictionary<int, int> parent = new Dictionary<int, int>();